server/utils/appendFieldUtils.ts

```diff
@@ -0,0 +1,26 @@
+import { Request } from 'express'
+import { formatDisplayDateTime } from './datetimeUtils'
+import { lastNameCommaFirstName } from './miniProfileUtils'
+import Prisoner from '../services/prisonerSearch/prisoner'
+
+/**
+ * If the description + divider test is equal to or more than 3000 characters, then show the character count
+ * If the description + divider text is less than 3000 characters, do not show the character count until the user reaches a total of 3000 characters for all entries plus dividers
+ */
+export const generateSaveTimestamp = (prisoner: Pick<Prisoner, 'firstName' | 'lastName'>) =>
+  `\n\n[On ${formatDisplayDateTime(new Date().toString())}, ${lastNameCommaFirstName(prisoner)} added:]\n\n`
+
+export const getMaxCharsAndThreshold = (req: Request, appendField?: string) => {
+  const timestampLength = generateSaveTimestamp(req.journeyData.prisoner!).length
```

Comment:
This should be the display name of the logged in user, not the prisoner's name. [On time and date, userLastname, userFirstname added:]